components/Themes/src/MDCFontScheme.h

```diff
@@ -0,0 +1,90 @@
+/*
+ Copyright 2017-present the Material Components for iOS authors. All Rights Reserved.
+
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+ http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+ */
+
+#import <UIKit/UIKit.h>
+
+/**
+ A color scheme comprised of set of primary and secondary colors. Material design guidelines
+ recommend using primary and secondary colors with light and dark color variants.
+ */
+@protocol MDCFontScheme <NSObject>
+
+/** The font-face to be used for Display 1. */
+@property(nonatomic, strong, nullable, readonly) UIFont *display1;
```

Comment:
I think they should be nonnull so the themer doesn't have to rely on its own defaults.

Reply:
This is by design. Consider our implementations use customFont = nil as 'default to system'.

So the code for our hypothetical FooThemers becomes:
// Foo has two text elements
// A quote that should be body2
// An author that should be caption
MDCFooComponentThemer
foo.quoteFont = scheme.body2  // If nil this will use the system font
foo.author = scheme.caption      // If nil this will use the system font

Reply:
lets comment that these are setting on component properties that are in spirit null resetable .

Reply:
Noted.